src/main/java/org/sonarlint/intellij/ui/ReportPanel.java

```diff
@@ -143,15 +144,19 @@ public void updateFindings(AnalysisResult analysisResult) {
         .filter(e -> !e.getValue().isEmpty())
         .collect(Collectors.toMap(Map.Entry::getKey, e -> (Collection<LiveIssue>) e.getValue()));
 
-      treeBuilder.updateModel(newIssues);
-      oldTreeBuilder.updateModel(oldIssues);
-      securityHotspotTreeBuilder.updateModel(newHotspots);
-      oldSecurityHotspotTreeBuilder.updateModel(oldHotspots);
+      runOnUiThread(project, () -> {
+        treeBuilder.updateModel(newIssues);
+        oldTreeBuilder.updateModel(oldIssues);
+        securityHotspotTreeBuilder.updateModel(newHotspots);
+        oldSecurityHotspotTreeBuilder.updateModel(oldHotspots);
+      });
     } else {
-      securityHotspotTreeBuilder.updateModel(findings.getSecurityHotspotsPerFile());
-      oldSecurityHotspotTreeBuilder.updateModel(Collections.emptyMap());
-      treeBuilder.updateModel(findings.getIssuesPerFile());
-      oldTreeBuilder.updateModel(Collections.emptyMap());
+      runOnUiThread(project, () -> {
```

Comment:
I think we can further optimize ui thread usage here, were are doing some computation inside update models

Reply:
It's quite difficult because it's computations on files, it breaks the logic if you change thread in the middle of everything. I tried and it caused issues, I'd prefer to go like this for now